### PR TITLE
Rename ws migrate to migrate-definition

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Rename `ws migrate` to `migrate-definition`.

### CODE REVIEW

File: tests/workspace/test_manager.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def test_migration_already_v2(
     project_directory,
 ):
     with project_directory("migration_already_v2"):
-        result = runner.invoke(["ws", "migrate"])
+        result = runner.invoke(["migrate-definition"])
```
Comment (sfc-gh-astus) [inline tests/workspace/test_manager.py:94]: These tests should be moved to tests/migrate/test_migrate_definition.py

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -39,7 +39,7 @@
 * Added support for external access (api integrations and secrets) in Streamlit.
 * Added support for `<% ... %>` syntax in SQL templating.
 * Support multiple Streamlit application in single snowflake.yml project definition file.
-* Added `snow ws migrate` command to migrate `snowflake.yml` file from V1 to V2.
+* Added `snow migrate-defintion` command to migrate `snowflake.yml` file from V1 to V2.
```
Comment (sfc-gh-bdufour) [inline RELEASE-NOTES.md:42]: typo

File: src/snowflake/cli/_app/commands_registration/builtin_plugins.py [hunk 3]
```diff
@@ -32,6 +33,7 @@ def get_builtin_plugin_name_to_plugin_spec():
     plugin_specs = {
         "connection": connection_plugin_spec,
         "spcs": spcs_plugin_spec,
+        "migrate-definition": migrate_plugin_spec,
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/_app/commands_registration/builtin_plugins.py:36]: Not sure "definition" is clear here from context. Also, this feels odd as a top level command. Would it make sense to have some kind of grouping for these "utility" commands? e.g. if we wanted to add a command to validate the PDF file (might be a nice addition to support migration efforts). There would be essentially no good home for this.

File: RELEASE-NOTES.md (hunk 2 above)
Comment (sfc-gh-mraba) [inline RELEASE-NOTES.md:42]:
```suggestion
* Added `snow migrate-definition` command to migrate `snowflake.yml` file from V1 to V2.
```

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**